secretmanager/api/SecretManagerSample/Program.cs

```diff
@@ -0,0 +1,381 @@
+        public string name { get; set; }
+    }
+
+    [Verb("create", HelpText = "Create a secret")]
+    class CreateSecretOptions
+    {
+        [Value(0, HelpText = "Full resource name of project", Required = true)]
+        public string projectId { get; set; }
+
+        [Value(1, HelpText = "Name of the secret", Required = true)]
+        public string id { get; set; }
+    }
+
+    [Verb("delete", HelpText = "Delete secret")]
+    class DeleteSecretOptions : SecretOptions { }
+
+    [Verb("get", HelpText = "Get secret")]
+    class GetSecretOptions : SecretOptions { }
+
+    [Verb("list", HelpText = "List secrets")]
+    class ListSecretsOptions : SecretOptions { }
+
+    [Verb("update", HelpText = "Update secret")]
+    class UpdateSecretOptions : SecretOptions { }
+
+    [Verb("access-version", HelpText = "Access the provided secret version")]
+    class AccessSecretVersionOptions : SecretVersionOptions { }
+
+    [Verb("add-version", HelpText = "Add a new version")]
+    class AddSecretVersionOptions : SecretVersionOptions { }
+
+    [Verb("destroy-version", HelpText = "Destroy secret version")]
+    class DestroySecretVersionOptions : SecretVersionOptions { }
+
+    [Verb("disable-version", HelpText = "Disable secret version")]
+    class DisableSecretVersionOptions : SecretVersionOptions { }
+
+    [Verb("enable-version", HelpText = "Enable secret version")]
+    class EnableSecretVersionOptions : SecretVersionOptions { }
+
+    [Verb("get-version", HelpText = "Get secret version")]
+    class GetSecretVersionOptions : SecretVersionOptions { }
+
+    [Verb("list-versions", HelpText = "List secret versions")]
+    class ListSecretVersionsOptions : SecretVersionOptions { }
+
+
+    public class SecretManagerSample
+    {
+        // [START secretmanager_access_secret_version]
+        public static void AccessSecretVersion(string name)
+        {
+            // string name = "projects/my-project/secrets/my-secret/versions/5";
+            // string name = "projects/my-project/secrets/my-secret/versions/latest";
+
+            SecretManagerServiceClient client = SecretManagerServiceClient.Create();
+
+            // Create the request.
+            var request = new AccessSecretVersionRequest
+            {
+                Name = name,
+            };
+
+            // Access the secret and print the result.
+            //
+            // WARNING: Do not print secrets in production environments. This
+            // snippet is for demonstration purposes only.
+            var version = client.AccessSecretVersion(request);
+            string payload = version.Payload.Data.ToStringUtf8();
+            Console.WriteLine($"Payload: {payload}");
+        }
+        // [END secretmanager_access_secret_version]
+
+        // [START secretmanager_add_secret_version]
+        public static void AddSecretVersion(string parent)
+        {
+            // string parent = "projects/my-project/secrets/my-secret";
```

Comment:
Instead of using this commented line here (if uncommented, this won't compile), prefer using parameters with default values or better still XML documentation for the method.

`public static void AddSecretVersion(string parent = "projects/my-project/secrets/my-secret")`

```
/// <summary>
/// ...
/// </summary>
/// <param name="parent">....</param>
```